compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2084,6 +2091,10 @@ public void visit(BLangRecordLiteral recordLiteral) {
                 }
             }
         }
+
+        if (isInferredRecordForMapCET) {
+            recordLiteral.expectedType = type;
```

Comment:
Why is the expected type changed here?

Reply:
We don't log the error at L2079 if a record type is inferred for a mapping constructor expression when the contextually expected type is a map type.

e.g.,
```ballerina
map<int> m = {
   readonly a: 1,
   b: 2
};
```

Here the actual type of the constructed value is `record {| readonly int a; int...; |}`. Usually you'll have to use a literal key to provide for the rest field (`b` in the sample), but that is not required when a record type is inferred for a map ACET.

To do this, we need to know here what the original applicable expected type was.

Usually, the `expectedType` field of an expression is overridden in `checkExpr`, but for this particular scenario we don't do it because we need to know the exact ACET - see https://github.com/ballerina-platform/ballerina-lang/blob/ed729d2aa3f90d27522bb8d1f867a4ee61da6cd8/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L378

I don't think this is required, but just for uniformity we set the `expectedType` as the normal path here.